_posts/2015-08-15-Cocoa-Architecture:-Router-Pattern.markdown

```diff
@@ -0,0 +1,164 @@
+---
+layout: post
+title: "Cocoa Architecture: ARRouter"
+date: 2015-08-15
+comments: false
+categories: [ios, mobile, architecture, energy, eigen, eidolon, hal]
+author: Orta Therox
+github-url: https://www.github.com/orta
+twitter-url: http://twitter.com/orta
+blog-url: http://orta.io
+---
+
+I want to talk about a pattern that we've been using for the last few years on the Artsy Mobile team. This pattern pre-dates me joining Artsy by a few weeks, and was introduced into our codebase by Ben Jackson, this was it's first method:
+
+```objc
+  + (NSURL *)newOAuthURLWithUsername:(NSString *)username password:(NSString *)password {
+      NSDictionary *params = [[NSDictionary alloc] initWithObjectsAndKeys:
```

Comment:
I'd forgotten about this method! Burnt me so many times when I switched key/value order.